### PR TITLE
Elixir: recognize if constructs, func def, and module def

### PR DESCRIPTION
**Pull request by aryx**
test plan:
```
elixir_if/home/pad/yy/tests/parsing/elixir $ yy -l elixir -dump_elixir_raw_ast if2.ex
+ /home/pad/yy/_build/default/src/main/Main.exe -l elixir -dump_elixir_raw_ast if2.ex
[(Call
    ((I (Id ("if", ()))), ((), ([(I (Id ("foo", ())))], []), ()),
     (Some ((), ((Body [(I (Id ("bar", ())))]), []), ()))))
  ]
elixir_if/home/pad/yy/tests/parsing/elixir $ yy -l elixir -dump_elixir_ast if2.ex
+ /home/pad/yy/_build/default/src/main/Main.exe -l elixir -dump_elixir_ast if2.ex
[(S (If ((), (I (Id ("foo", ()))), (), [(I (Id ("bar", ())))], None, ())))]
elixir_if/home/pad/yy/tests/parsing/elixir $ yy -l elixir -dump_ast if2.ex
+ /home/pad/yy/_build/default/src/main/Main.exe -l elixir -dump_ast if2.ex
Pr(
  [If((),
     Cond(
       N(
         Id(("foo", ()),
           {id_info_id=1; id_flags=Ref(0); id_resolved=Ref(None); id_type=Ref(None); id_svalue=Ref(None); }))),
     Block(
       [ExprStmt(
          N(
            Id(("bar", ()),
              {id_info_id=2; id_flags=Ref(0); id_resolved=Ref(None); id_type=Ref(None); id_svalue=Ref(None); })), ())]),
     None)])
elixir_if/home/pad/yy/tests/parsing/elixir $
```

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by aryx** [comment]
using visitors ppx!!

File: tests/parsing/elixir/module.ex [hunk 1]
```diff
@@ -1,8 +1,4 @@
-defmodule MyAppWeb.UserController do
-  use MyAppWeb, :controller
-
-  def show(conn, %{"id" => id}) do
-    user = Repo.get(User, id)
-    render(conn, :show, user: user)
-  end
+defmodule Number do
```
Comment (IagoAbal) [inline tests/parsing/elixir/module.ex:1]: These tests are just checking that the code can be parsed, so it doesn't really test the elixir-to-generic conversion, right?

Comment (IagoAbal) [reply tests/parsing/elixir/module.ex:1]: I guess we don't have that kind of test suite yet (we could write such tests using something like Cram perhaps). But I'll add some taint maturity tests in a subsequent PR.

**Review by IagoAbal — APPROVED** [review]

File: tests/parsing/elixir/module.ex (hunk 1 above)
Comment (aryx) [reply tests/parsing/elixir/module.ex:1]: correct, at some point we should have tainting test or matching tests that exercise those code paths.

**Pull request merged**